emeis/core/models.py

```diff
@@ -1,7 +1,127 @@
-from django.contrib.auth.models import AbstractUser
+import unicodedata
+import uuid
 
-# from django.db import models
+from django.contrib.auth.validators import UnicodeUsernameValidator
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
 
 
-class User(AbstractUser):
-    pass
+class BaseModel(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True, db_index=True)
+    modified_at = models.DateTimeField(auto_now=True, db_index=True)
+    created_by_user = models.CharField(
+        max_length=150, blank=True, null=True, db_index=True
+    )
+    created_by_group = models.CharField(
+        max_length=150, blank=True, null=True, db_index=True
+    )
+
+    class Meta:
+        abstract = True
+
+
+class UUIDModel(BaseModel):
+    """
+    Models which use uuid as primary key.
+
+    Defined as Caluma default
+    """
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    class Meta:
+        abstract = True
+
+
+class User(UUIDModel):
+    username_validator = UnicodeUsernameValidator()
+
+    username = models.CharField(
+        _("username"),
+        max_length=150,
+        unique=True,
+        help_text=_(
+            "Required. 150 characters or fewer. Letters, digits and @/./+/-/_ only."
+        ),
+        validators=[username_validator],
+        error_messages={"unique": _("A user with that username already exists.")},
+    )
+    first_name = models.CharField(_("first name"), max_length=30, blank=True)
```

Comment:
I wouldn't artificially limit the first name. While it's unlikely in our countries, in other regions it might be normal to have a longer first name

Reply:
Totally agree. I've removed the restictions on all names (username, first and last).